Bugsnag/Breadcrumbs/BSGNotificationBreadcrumbs.m

```diff
@@ -0,0 +1,304 @@
+        
+        NSWindowDidBecomeKeyNotification,
+        NSWindowDidEnterFullScreenNotification,
+        NSWindowDidExitFullScreenNotification,
+        NSWindowWillCloseNotification,
+        NSWindowWillMiniaturizeNotification,
+    ];
+#endif
+    return nil;
+}
+
+- (NSArray<NSNotificationName> *)automaticBreadcrumbControlEvents {
+#if TARGET_OS_IOS
+    return @[
+        UITextFieldTextDidBeginEditingNotification,
+        UITextFieldTextDidEndEditingNotification,
+        UITextViewTextDidBeginEditingNotification,
+        UITextViewTextDidEndEditingNotification
+    ];
+#elif TARGET_OS_OSX
+    return @[
+        NSControlTextDidBeginEditingNotification,
+        NSControlTextDidEndEditingNotification
+    ];
+#endif
+    return nil;
+}
+
+- (NSArray<NSNotificationName> *)automaticBreadcrumbTableItemEvents {
+#if TARGET_OS_IOS || TARGET_OS_TV
+    return @[ UITableViewSelectionDidChangeNotification ];
+#elif TARGET_OS_OSX
+    return @[ NSTableViewSelectionDidChangeNotification ];
+#endif
+    return nil;
+}
+
+- (NSArray<NSNotificationName> *)automaticBreadcrumbMenuItemEvents {
+#if TARGET_OS_OSX
+    return @[ NSMenuWillSendActionNotification ];
+#endif
+    return nil;
+}
+
+- (void)dealloc {
+    [_notificationCenter removeObserver:self];
+}
+
+#pragma mark -
+
+- (NSString *)messageForNotificationName:(NSNotificationName)name {
+    return self.notificationNameMap[name] ?: [name stringByReplacingOccurrencesOfString:@"Notification" withString:@""];
+}
+
+- (void)addBreadcrumbWithType:(BSGBreadcrumbType)type forNotificationName:(NSNotificationName)notificationName {
+    [self addBreadcrumbWithType:type forNotificationName:notificationName metadata:nil];
+}
+
+- (void)addBreadcrumbWithType:(BSGBreadcrumbType)type forNotificationName:(NSNotificationName)notificationName metadata:(NSDictionary *)metadata {
+    [self.breadcrumbSink leaveBreadcrumbWithMessage:[self messageForNotificationName:notificationName] metadata:metadata ?: @{} andType:type];
+}
+
+#pragma mark -
+
+- (void)start {
+    // State events
+    if ([_configuration shouldRecordBreadcrumbType:BSGBreadcrumbTypeState]) {
+        // Generic state events
+        for (NSNotificationName name in [self automaticBreadcrumbStateEvents]) {
+            [self startListeningForStateChangeNotification:name];
+        }
+        
+#if TARGET_OS_OSX
+        // Workspace-specific events - MacOS only
+        for (NSNotificationName name in [self workspaceBreadcrumbStateEvents]) {
+            NSLog(@"%@", name);
+            [_workspaceNotificationCenter addObserver:self
+                                             selector:@selector(addBreadcrumbForNotification:)
+                                                 name:name
+                                               object:nil];
+        }
+#endif
+        
+        // NSMenu events (Mac only)
+        for (NSNotificationName name in [self automaticBreadcrumbMenuItemEvents]) {
```

Comment:
should this `if TARGET_OS_OSX` conditional enclose the mac only bits in this loop?

Reply:
Probably, yes; much of this is the original copy and pasted logic!

Reply:
Done 👍